pkg/engine/imageVerify.go

```diff
@@ -25,6 +25,61 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
+func hasMatchedImages(images map[string]map[string]apiutils.ImageInfo, rule *kyvernov1.Rule) bool {
+	count := 0
+	for _, infoMap := range images {
+		for _, imageInfo := range infoMap {
+			image := imageInfo.String()
+			for _, verifyImage := range rule.VerifyImages {
+				verifyImage = *verifyImage.Convert()
+				if imageMatches(image, verifyImage.ImageReferences) {
+					count += 1
+				}
+			}
+		}
+	}
+	return count > 0
+}
+
+func hasVariables(verifyImages []kyvernov1.ImageVerification) bool {
+	for _, verifyImage := range verifyImages {
+		verifyImage = *verifyImage.Convert()
+		for _, imageRef := range verifyImage.ImageReferences {
+			matches := variables.RegexVariables.FindAllString(imageRef, -1)
+			if len(matches) > 0 {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func requireValidation(policyContext *PolicyContext, rule *kyvernov1.Rule) (bool, error) {
+	var (
+		images map[string]map[string]apiutils.ImageInfo
+		err    error
+	)
+	variables := hasVariables(rule.VerifyImages)
```

Comment:
Can we perform this check during policy validation instead?

Reply:
if we do this in policy validation, wouldn't it block passing variables to imageRef altogether?

Reply:
Yes, the intent would be to not allow variables in the image references.

Reply:
fixed